src/AggregateFunctions/ReservoirSampler.h

```diff
@@ -161,6 +161,13 @@ class ReservoirSampler
     {
         if (sample_count != b.sample_count)
             throw Poco::Exception("Cannot merge ReservoirSampler's with different sample_count");
+
+        // There will be an aliasing issue if we merge the same object with itself. I.e. we will insert from `b.samples` into `a.samples`,
+        // but both refer to the same array. It might happen in case of multiplying an aggregate function state by a numberic constant.
```

Comment:
```suggestion
        // but both refer to the same array. It might happen in case of multiplying an aggregate function state by a numeric constant.
```